vcx/wrappers/node/src/rustlib.ts

```diff
@@ -9,12 +9,19 @@ export const VcxStatus = StructType({
   status: 'int'
 })
 
+interface IUintTypes {
+  [key: string]: string
+}
+const UINTS_TYPES: IUintTypes = { x86: 'uint32', x64: 'uint64' }
+const ARCHITECTURE: string = process.env.LIBVCX_FFI_ARCHITECTURE || 'x86'
+const FFI_UINT: string = UINTS_TYPES[ARCHITECTURE]
+
 // FFI Type Strings
 export const FFI_ERROR_CODE = 'int'
 export const FFI_BOOL = 'bool'
 export const FFI_CONNECTION_HANDLE = 'uint32'
 export const FFI_UNSIGNED_INT = 'uint32'
-export const FFI_UNSIGNED_INT_PTR = ref.refType('uint32')
```

Comment:
I don't much about `ffi` library, but `FFI_UNSIGNED_INT_PTR` was not actually used anywhere. So I've fixed signatures of `vcx_connection_sign_data ` and `vcx_connection_verify_signature ` to use this declaration, as they should according to Rust function signatures. 
However, if I change signatures and simply keep `FFI_UNSIGNED_INT_PTR = ref.refType('uint32')`, it doesn't work. When using `ref.refType`, sign function will fail like this:
```
Demo Faber: [error]: Unknown Error: error setting argument 2 - writePointer: Buffer instance expected as third argument
Demo Faber: [error]: Error: Unknown Error: error setting argument 2 - writePointer: Buffer instance expected as third argument
    at new VCXInternalError (/Users/prague/dev/hyperledger/indy-sdk/vcx/wrappers/node/dist/src/errors.js:10:23)
    at Connection._callee4$ (/Users/prague/dev/hyperledger/indy-sdk/vcx/wrappers/node/dist/src/api/connection.js:232:23)
    at tryCatch (/Users/prague/dev/hyperledger/indy-sdk/vcx/wrappers/node/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:65:40)
    at Generator.invoke [as _invoke] (/Users/prague/dev/hyperledger/indy-sdk/vcx/wrappers/node/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:303:22)
    at Generator.prototype.(anonymous function) [as throw] (/Users/prague/dev/hyperledger/indy-sdk/vcx/wrappers/node/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:117:21)
    at rejected (/Users/prague/dev/hyperledger/indy-sdk/vcx/wrappers/node/dist/src/api/connection.js:5:47)
    at <anonymous>
```
So I changed this line to 
```
export const FFI_UNSIGNED_INT_PTR = FFI_UINT
```
and with this, demo works fine.

Effectively nothing has changed. `ref.refType` was not previously used for the `vcx_connection_sign_data ` and `vcx_connection_verify_signature ` anyways. 
Whether it should be used, I don't know, is there a more `ffi` knowledgeable person around here?

Reply:
You can try out both cases with
`export const FFI_UNSIGNED_INT_PTR = ref.refType('uint32')`
and 
`export const FFI_UNSIGNED_INT_PTR = FFI_UINT`
By running new demo files, the commands to run them are added to `package.json`